### ISSUE TITLE
thchs30 decode error

### ISSUE THREAD

**Issue by Mitomzhou** [athena-team/athena#112]
when I run thchs30 (like example aishell), I got an error during decoding ,  but  procedure Fine-turning and Training language model are successful.
error message:

```
 ERROR:tensorflow:
Object was never used (type <class 'tensorflow.python.ops.tensor_array_ops.TensorArray'>):
<tensorflow.python.ops.tensor_array_ops.TensorArray object at 0x7f77941ae910>
If you want to mark it as used call its "mark_used()" method.
It was originally created here:
  File "athena/decode_main.py", line 89, in <module>
    decode(jsonfile, n=5, log_file='nohup_thchs30.out')  
File "athena/decode_main.py", line 73, in decode
    solver.decode(dataset_builder.as_dataset(batch_size=1))  
File "/raid/BH/mitom/athena/athena/solver.py", line 149, in decode
    predictions = self.model.decode(samples, self.hparams, lm_model=self.lm_model)  
File "/raid/BH/mitom/athena/athena/models/mtl_seq2seq.py", line 109, in decode 
  history_predictions.write(0, last_predictions) 
File "/home/bh/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/util/tf_should_use.py", line 237, in wrapped
    error_in_function=error_in_function)

```

**Comment by Some-random** [comment]
We haven't seen errors like this before. What's your TF version? Does this error occur on every speech  you try to decode?

**Comment by Mitomzhou** [comment]
> We haven't seen errors like this before. What's your TF version? Does this error occur on every speech you try to decode?
-----------------------------------------------------------------------
Thanks for your reply.
I havent't seen error before. I run aishell decoder successfully, and got acc 91%.  The error occur on thchs30 decoder, then got acc 0.01% , but procedure Fine-turning and Training language model  both got acc >90%.  I don't know what happen?  my wav data preparing error? 
env:  
OS: mint18.3 (ubuntu)
Tensorflow-gpu==2.1.0  
python=3.5
Single GPU Tesla v100

**Comment by Some-random** [comment]
Does this error occur on every audio in test set of thchs30 or only some specific audio?

**Comment by gongstar1** [comment]
I got some issue, modified as below, it works, may be the tensorArray is not sync.
history_predictions = history_predictions.write(0, last_predictions)

**Comment by Mitomzhou** [comment]
> I got some issue, modified as below, it works, may be the tensorArray is not sync.
> history_predictions = history_predictions.write(0, last_predictions)

I solved the problem with your advice. Thanks.

**Comment by LeoMax-Xiong** [comment]
> when I run thchs30 (like example aishell), I got an error during decoding , but procedure Fine-turning and Training language model are successful.
> error message:
> 
> ```
>  ERROR:tensorflow:
> Object was never used (type <class 'tensorflow.python.ops.tensor_array_ops.TensorArray'>):
> <tensorflow.python.ops.tensor_array_ops.TensorArray object at 0x7f77941ae910>
> If you want to mark it as used call its "mark_used()" method.
> It was originally created here:
>   File "athena/decode_main.py", line 89, in <module>
>     decode(jsonfile, n=5, log_file='nohup_thchs30.out')  
> File "athena/decode_main.py", line 73, in decode
>     solver.decode(dataset_builder.as_dataset(batch_size=1))  
> File "/raid/BH/mitom/athena/athena/solver.py", line 149, in decode
>     predictions = self.model.decode(samples, self.hparams, lm_model=self.lm_model)  
> File "/raid/BH/mitom/athena/athena/models/mtl_seq2seq.py", line 109, in decode 
>   history_predictions.write(0, last_predictions) 
> File "/home/bh/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/util/tf_should_use.py", line 237, in wrapped
>     error_in_function=error_in_function)
> ```

I meet many errors with tensorflow==2.1.0, But with little errors with tensorflow==2.0.1 as the requirements。

**Comment by Some-random** [comment]
This should be a bug on our end. It seems like TensorArray.write did a default copy under tf 2.0.1 (which is what we intended to do) and this behavior is fixed for tf 2.1.0. We will make a pr to fix it.
